src/harness/testcases/WINNF_FT_S_DRG_testcase.py

```diff
@@ -423,7 +425,12 @@ def test_WINNF_FT_S_DRG_6(self, config_filename):
           self.assertEqual(response['cbsdId'], deregister_request[i]['cbsdId'])
         else:
           self.assertFalse('cbsdId' in response)
-
+      # remove still registered CBSDs from the next preload conditional registration
+      if response['response']['responseCode'] != 0:
+        cbsd_to_be_removed_from_next_reg = registration_request[i]
+        indexes_conditional_to_remove = [index for index, c in enumerate(conditional_registration_data) if c['cbsdSerialNumber'] == cbsd_to_be_removed_from_next_reg['cbsdSerialNumber'] and c['fccId'] == cbsd_to_be_removed_from_next_reg['fccId']]
+        if indexes_conditional_to_remove:
+          del conditional_registration_data[indexes_conditional_to_remove[0]]
```

Comment:
If for some reason the config file has multiple conditional reg entries for the same (FCC ID, SN) pair, this will only remove the first. You probably want something like

```
for index in reversed(indexes_conditional_to_remove):
  del conditional_registration_data[index]
```
Please double-check the logic though to make sure that it works correctly. Note specifically the reversal of the list (taking advantage of the fact that it's already sorted in ascending order) so that we don't change the meaning of the index while using it.